lib/bolt/catalog.rb

```diff
@@ -74,7 +73,8 @@ def compile_catalog(request)
                                        facts: target["facts"] || {},
                                        variables: target["variables"] || {}) do |pal|
           Puppet.override(bolt_pdb_client: pdb_client,
-                          bolt_inventory: setup_inventory(request['inventory'])) do
+                          bolt_inventory: setup_inventory(request['inventory']),
+                          bolt_executor: nil) do
```

Comment:
@nicklewis For plan functions that are allowed in apply blocks, the default `nil` block handled the case where we look up the executor for analytics. Should we handle that by overriding in catalog compilation? Or should we keep the default `nil` blocks in the various plan functions?

Reply:
If there's a case where it's _legal_ for it to be `nil`, then we should have a default block and appropriately handle it being `nil`. But for cases where that _isn't_ supposed to be possible, we should just let it fail rather than having a default and then failing explicitly.